digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataImporter/MassTransitStopCsvImporter.scala

```diff
@@ -0,0 +1,438 @@
+import org.apache.commons.lang3.StringUtils.isBlank
+
+import scala.util.Try
+
+class MassTransitStopCsvOperation(vvhClientImpl: VVHClient, roadLinkServiceImpl: RoadLinkService, eventBusImpl: DigiroadEventBus) {
+  lazy val propertyUpdater = new Updater(vvhClientImpl: VVHClient, roadLinkServiceImpl: RoadLinkService, eventBusImpl: DigiroadEventBus)
+  lazy val positionUpdater = new PositionUpdater(vvhClientImpl: VVHClient, roadLinkServiceImpl: RoadLinkService, eventBusImpl: DigiroadEventBus)
+  lazy val creator = new Creator(vvhClientImpl: VVHClient, roadLinkServiceImpl: RoadLinkService, eventBusImpl: DigiroadEventBus)
+
+  private def getStrategies(): Seq[CsvOperations] = {
+    Seq(propertyUpdater, creator, positionUpdater)
+  }
+
+  def getStrategy(csvRowWithHeaders: Map[String, String]): CsvOperations = {
+    val strategies = getStrategies()
+    strategies.find(strategy => strategy.is(csvRowWithHeaders)).getOrElse(throw new UnsupportedOperationException(s"Please check the combination between Koordinaatti and  Valtakunnallinen ID"))
+  }
+
+  def importAssets(inputStream: InputStream, fileName: String, user: User, logId: Long, roadTypeLimitations: Set[AdministrativeClass]): Unit = {
+    val streamReader = new InputStreamReader(inputStream, "UTF-8")
+    val csvReader = CSVReader.open(streamReader)(new DefaultCSVFormat {
+      override val delimiter: Char = ';'
+    })
+    val csvRead = csvReader.allWithHeaders()
+    val csvRow = csvRead.map {
+      r => r.toList.map ( p => (p._1.toLowerCase, p._2)).toMap
+    }
+
+    val strategy = getStrategy(csvRow.head)
+    strategy.importAssets(csvRow: List[Map[String, String]], fileName, user, logId, roadTypeLimitations)
+    }
+}
+
+
+trait MassTransitStopCsvImporter extends PointAssetCsvImporter {
+  def roadLinkService: RoadLinkService
+  def eventBus: DigiroadEventBus
+
+  class AssetNotFoundException(externalId: Long) extends RuntimeException
+
+  type ExcludedRoadLinkTypes = List[AdministrativeClass]
+
+  val massTransitStopService: MassTransitStopService = {
+    class MassTransitStopServiceWithDynTransaction(val eventbus: DigiroadEventBus, val roadLinkService: RoadLinkService, val roadAddressService: RoadAddressService) extends MassTransitStopService {
+      override def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+
+      override def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+
+      override val tierekisteriClient: TierekisteriMassTransitStopClient = tierekisteriMassTransitStopClient
+      override val massTransitStopDao: MassTransitStopDao = new MassTransitStopDao
+      override val municipalityDao: MunicipalityDao = new MunicipalityDao
+      override val geometryTransform: GeometryTransform = new GeometryTransform(roadAddressService)
+    }
+    new MassTransitStopServiceWithDynTransaction(eventBus, roadLinkService, roadAddressService)
+  }
+  override def mandatoryFieldsMapping: Map[String, String]
```

Comment:
I am getting a warning on this in the IDE. Solved with
`override def mandatoryFieldsMapping: Map[String, String] = Map()`